File: src/main/java/com/musinsa/subject/entity/Brand.java
package com.musinsa.subject.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;

import java.io.Serializable;

@Entity
@Table(name = "brand")
@DynamicInsert
@DynamicUpdate
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Brand extends BaseTimeEntity implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "brand_id")
    private Long brandId;

    @Column(name = "brand_name", nullable = false, length = 100)
    private String brandName;

}

File: src/main/kotlin/com/musinsa/subject/entity/Brand.kt
package com.musinsa.subject.entity

import jakarta.persistence.*
import org.hibernate.annotations.DynamicInsert
import org.hibernate.annotations.DynamicUpdate
import java.io.Serializable

@Entity
@Table(name = "brand")
@DynamicInsert
@DynamicUpdate
class Brand(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "brand_id")
    var brandId: Long? = null,

    @Column(name = "brand_name", nullable = false, length = 100)
    var brandName: String
) : BaseTimeEntity(), Serializable
